src/sntp.rs

```diff
@@ -1,4 +1,8 @@
+//! SNTP Time Synchronization
+
 use core::cmp::min;
+use core::time::Duration;
+use std::sync::Mutex;
```

Comment:
Sorry, one more thing that you need to fix: please use the Mutex from `crate::private::mutex::Mutex`. It allows folks to use `esp-idf-svc` in no_std environments as well (not that there is much of a use case for that though).

You can see how to use that mutex in the other modules, i.e. `EspNow`, `Eth` etc. Or just look at the mutex used by the `TAKEN` static var.